File: cfgs/java_cfgs_foo/MODULAR_EXPONENTIATION_POWER_IN_MODULAR_ARITHMETIC.java
Nodes
1;special;;;
2;statement;int res = 1;14;
3;statement;x = foo_1 % foo_2;15;
4;conditional;foo_1 > 0;16;
5;conditional;(foo_1 & 1) == 1;17;
6;statement;res = (foo_1 * foo_2) % foo_3;17;
7;statement;y = foo_1 >> 1;18;
8;statement;x = (foo_1 * foo_2) % foo_3;19;
9;statement;return foo_1;21;
10;exit;;;
Edges
0;2;
1;3;
2;4;
3;9;5;
4;6;7;
5;7;
6;8;
7;4;
8;10;
9;

File: cfgs/pythoncfgs/MODULAR_EXPONENTIATION_POWER_IN_MODULAR_ARITHMETIC.py
Nodes
1;statement;foo_1 = 1;8;
2;statement;foo_1 = foo_2 % foo_3;9;
3;conditional;foo_1 > 0;10;
4;statement;return foo_1;15;
5;conditional;foo_1 & 1 == 1;11;
6;statement;foo_1 = foo_2 >> 1;13;
7;statement;foo_1 = foo_2 * foo_3 % foo_4;14;
8;statement;foo_1 = foo_2 * foo_3 % foo_4;12;
9;exit;;
Edges
0;2;
1;3;
2;4;5;
3;9;
4;6;8;
5;7;
6;3;
7;6;
8;